<?xml version="1.0"?>
<launch>
    <!-- start basic args -->
    <arg name="veh" default="$(env VEHICLE_NAME)"/>
    <arg name="inference_node_name" default="road_model_inference"/>
    <arg name="reaction_node_name" default="road_inference_to_reaction"/>
    <arg name="pkg_name" default="road_following"/>   
    <arg name="config" default="baseline" />
    <arg name="reaction_output" default="screen" />
    <arg name="confidence_threshold" default="0.8"/>
    <arg name="teleop" default="false"/>
    
    <!-- start switch args -->
    <arg name="raw" default="true"/>
    <arg name="reaction" default="false"/>
    <arg name="camera" default="true"/>

    <!-- jetson_camera -->
    <group if="$(arg camera)">
        <include file="$(find jetson_camera)/launch/jetson_camera.launch">
		    <arg name="veh" value="$(arg veh)"/>
		    <arg name="config" value="$(arg config)"/>
            <arg name="raw" value="$(arg raw)"/>    
        </include>

    </group>

    <!-- inference model -->
    <group ns="$(arg veh)">
        <remap from="/$(arg veh)/$(arg inference_node_name)/image/raw" to="/$(arg veh)/jetson_camera/image/raw"/>
        <node  pkg="$(arg pkg_name)" type="$(arg inference_node_name).py" name="$(arg inference_node_name)" clear_params="true" output="screen" required="true">
            <rosparam file="$(find road_following)/param/inference_model.yaml" command="load"/>
        </node>
    </group> <!-- group ns -->

    <group if="$(arg reaction)">
        <remap from="/$(arg veh)/$(arg reaction_node_name)/inference" to="/$(arg veh)/$(arg inference_node_name)/inference"/>       
        <node ns="$(arg veh)" pkg="$(arg pkg_name)" type="$(arg reaction_node_name).py" name="$(arg reaction_node_name)" clear_params="true" output="$(arg reaction_output)" required="true"/>

        <!-- run inverse_kinematics_node -->
        <remap from="inverse_kinematics_node/car_cmd" to="$(arg reaction_node_name)/car_cmd"/>
        <remap from="inverse_kinematics_node/wheels_cmd" to="wheels_driver_node/wheels_cmd" />
        <include file="$(find ominibot_car)/launch/inverse_kinematics_node.launch">
            <arg name="veh" value="$(arg veh)"/>
            <arg name="config" value="$(arg config)"/>
        </include>

        <!-- Wheels Driver -->
        <include file="$(find ominibot_car)/launch/wheels_ominibot_driver_node.launch">
            <arg name="veh" value="$(arg veh)"/>
        </include>

    </group> <!-- group if=$(arg reaction) -->

</launch>